src/plat/pc99/machine/ioapic.c

```diff
@@ -52,12 +57,23 @@ static uint32_t ioapic_read(uint32_t ioapic, word_t reg)
 static void single_ioapic_init(word_t ioapic, cpu_id_t delivery_cpu)
 {
     uint32_t i;
+    uint32_t nirqs;
+
+    ioapic_write(ioapic, IOAPIC_REGSEL, IOAPIC_REG_IOAPICVER);
+    nirqs = (ioapic_read(ioapic, IOAPIC_WINDOW) >> 16) + 1;
+    ioapic_nirqs[ioapic] = nirqs;
+
+    /*
+     * All current implementations have 24 or fewer lines
+     * Protect against the future one that may have more
+     */
+    assert(nirqs <= IOAPIC_IRQ_LINES);
```

Comment:
`assert` will not be running in release configs, and verification does need to know that all values in that array are bounded. It looks like this is kernel init code, so if there is a condition where the hardware does not meet expectations, it should properly abort initialisation and bail out, not assert.

Reply:
OK, I'll do something different.  I think it's important at this point that we do bail out.

Reply:
OK I think this is fixed now.